packages/ui/src/organisms/OutOfStock/OutOfStock.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react'
+import type { ReactNode, FormHTMLAttributes } from 'react'
+
+import Form from '../../molecules/Form'
+
+export type OutOfStockProps = {
+  /**
+   * ID to find this component in testing tools (e.g.: cypress,
+   * testing-library, and jest).
+   */
+  testId?: string
+  /**
+   * The Out of Stock Section's title.
+   */
+  title: string | ReactNode
+  /**
+   * Message describing when the user will be notified.
+   */
+  message?: string | ReactNode
+  /**
+   *
+   */
+  onSubmit?: (event: React.FormEvent<HTMLFormElement>) => void
+  /**
+   *
+   */
+  children: ReactNode
+} & Omit<FormHTMLAttributes<HTMLFormElement>, 'title'>
+
+function OutOfStock({
+  testId = 'store-out-of-stock',
+  title,
+  message,
+  children,
+  ...formProps
+}: OutOfStockProps) {
+  return (
+    <section data-store-out-of-stock data-testid={testId} aria-live="polite">
+      <Form
+        data-store-out-of-stock-form
+        testId={`${testId}-form`}
+        {...formProps}
+      >
+        {typeof title === 'string' ? (
+          <p data-store-out-of-stock-title data-testid={`${testId}-title`}>
+            {title}
+          </p>
+        ) : (
+          <div data-store-out-of-stock-title data-testid={`${testId}-title`}>
+            {title}
+          </div>
+        )}
```

Comment:
_Broader question:_
Interesting approach. It feels a bit verbose if we must do this every time we have a prop `string | ReactNode`. Not sure there's a better way to handle this though, so I'm fine with it.

Reply:
Not sure there's a better way as well.

Thinking out loud: Another thing is that we are adding a wrapper `div` every time we pass a `ReactNode`. So we will have always a nested structure and +1 DOM Node.

```js
<div data-store-out-of-stock-title ... >
  <h1> title example </h1>
</div>
```

Reply:
Yeah... I know this problem, but it's an approach used in others components (Create a wrapper div with the data-attribute), I don't like too, but couldn't think another possibility.

The only thing which I can think is just render the component, without data-attributes, since if a developer chose to use a component as title/message maybe they don't need the data-attributes and the styles.